### PR TITLE
Step2 - String calculator

### PR DESCRIPTION
**Pull request by Mingyum-Kim**
Hi Justin, nice to meet you again!
I've completed the Step 2 mission and submitted it for now! 🚀
Kotlin still feels a bit awkward to me... Looking forward to your review! :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,5 +2,14 @@
...
-- [x] Make `PersonTest` in Kotlin
\ No newline at end of file
+- [x] Make `PersonTest` in Kotlin
+
+## Step2 - String calculator
+
+- [x] Split user entered string input by signs '+','-','*','/'
+- [x] Remove space and extract number
+- [x] Implement calculator to make result
+- [x] Check edge case and fix errors
```
Comment (liquidjoo) [inline README.md:15]: I really like this requirements document 👍

File: src/test/kotlin/StringCalculatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            validateExpression(splitExpression)
+            var result = splitExpression.first().toInt()
+            for (i in 1 until splitExpression.size step 2) {
+                validateSign(splitExpression[i])
+                val sign = splitExpression[i]
+                validateNumber(splitExpression[i + 1])
+                val targetNumber = splitExpression[i + 1].toInt()
+                result = apply(result, sign, targetNumber)
+            }
+            return result;
```
Comment (liquidjoo) [inline src/test/kotlin/StringCalculatorTest.kt:17]: I feel like this method is doing too much. How about dividing its responsibilities?

File: src/test/kotlin/StringCalculatorTest.kt [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            }
+            if(sign == "-") {
+                return result - targetNumber
+            }
+            if(sign == "*") {
+                return result * targetNumber
+            }
+            if(sign == "/") {
+                return result / targetNumber
+            }
```
Comment (liquidjoo) [inline src/test/kotlin/StringCalculatorTest.kt:58]: I think having too many conditional statements isn't good for your code. How can we reduce them without using a switch statement?

File: src/test/kotlin/StringCalculatorTest.kt [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            .hasMessageContaining("Enter a number in the first input or last input")
+    }
+
+    @Test
+    fun `throw exception if there is consecutive sign in expression`() {
+        assertThatThrownBy {
+            StringCalculator("1 + + 1").calculate()
+        }.isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("Enter a valid expression")
+    }
```
Comment (liquidjoo) [inline src/test/kotlin/StringCalculatorTest.kt:115]: These exception test cases are really well-written 👍

**Review by liquidjoo — APPROVED** [review]
Hi Anna,
Your second mission was really good!
I left some comments on your PR.
I hope you review them and reflect on them in your next mission.
Please proceed to the next step accordingly.
Have a good one 😄
